src/Facility.CodeGen.JavaScript/JavaScriptGenerator.cs

```diff
@@ -820,32 +820,27 @@ private CodeGenOutput GenerateFastifyPluginOutput(ServiceInfo service)
 
 			return new CodeGenOutput(file);
 
-			string GetJsonSchemaType(string typeName)
+			string GetJsonSchemaType(ServiceTypeInfo? serviceType)
 			{
-				if (typeName.EndsWith("[]", StringComparison.Ordinal))
-					return $"{{ type: 'array', items: {GetJsonSchemaType(typeName.Substring(0, typeName.Length - 2))} }}";
-				if (typeName.StartsWith("nullable<", StringComparison.Ordinal))
-					return $"{{ oneOf: [ {GetJsonSchemaType(typeName.Substring(9, typeName.Length - 10))}, {{ type: 'null' }} ] }}";
-				if (typeName.StartsWith("result<", StringComparison.Ordinal))
-					return $"{{ type: 'object', properties: {{ value: {GetJsonSchemaType(typeName.Substring(7, typeName.Length - 8))}, error: {{ $ref: '_error' }} }} }}";
-				if (typeName.StartsWith("map<", StringComparison.Ordinal))
-					return $"{{ type: 'object', additionalProperties: {GetJsonSchemaType(typeName.Substring(4, typeName.Length - 5))} }}";
-				if (typeName is "error")
-					return $"{{ $ref: '{"_error"}' }}";
-				if (customTypes.Contains(typeName))
-					return $"{{ $ref: '{typeName}' }}";
-				if (typeName is "string" or "bytes" or "datetime")
-					return $"{{ type: '{"string"}' }}";
-				if (typeName is "boolean")
-					return $"{{ type: '{"boolean"}' }}";
-				if (typeName is "int32" or "int64")
-					return $"{{ type: '{"integer"}' }}";
-				if (typeName is "decimal" or "double")
-					return $"{{ type: '{"number"}' }}";
-				if (typeName is "object")
-					return $"{{ type: '{"object"}', additionalProperties: true }}";
-
-				throw new NotSupportedException("Unknown type " + typeName);
+				if (serviceType is null)
+					throw new ArgumentNullException(nameof(serviceType));
+
+				return serviceType.Kind switch
+				{
+					ServiceTypeKind.String or ServiceTypeKind.Bytes or ServiceTypeKind.DateTime or ServiceTypeKind.ExternalEnum => $"{{ type: '{"string"}' }}",
+					ServiceTypeKind.Boolean => $"{{ type: '{"boolean"}' }}",
+					ServiceTypeKind.Double or ServiceTypeKind.Decimal => $"{{ type: '{"number"}' }}",
+					ServiceTypeKind.Int32 or ServiceTypeKind.Int64 => $"{{ type: '{"integer"}' }}",
+					ServiceTypeKind.Object or ServiceTypeKind.ExternalDto => $"{{ type: '{"object"}', additionalProperties: true }}",
+					ServiceTypeKind.Error => $"{{ $ref: '{"_error"}' }}",
+					ServiceTypeKind.Dto => $"{{ $ref: '{serviceType.Dto!.Name}' }}",
+					ServiceTypeKind.Enum => $"{{ $ref: '{serviceType.Enum!.Name}' }}",
+					ServiceTypeKind.Result => $"{{ type: 'object', properties: {{ value: {GetJsonSchemaType(serviceType.ValueType!)}, error: {{ $ref: '_error' }} }} }}",
```

Comment:
```suggestion
					ServiceTypeKind.Result => $$"""{ type: 'object', properties: { value: {{GetJsonSchemaType(serviceType.ValueType!)}}, error: { $ref: '_error' } } }",
```
I can't tell if this is better or not...

Reply:
You've got to double on braces somewhere I suppose. The current version is what I'm more familiar with as I don't see the `"""` syntax as much.